### PR TITLE
8282280: Update Xerces to Version 2.12.2

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Update to Xerces 2.12.2.

The update also fixes JDK-8144117. Added a test.

Tests: local XML tests passed. Tier2 passed. Two JCK tests failed with this update. See related issue report.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8282280](https://bugs.openjdk.java.net/browse/JDK-8282280): Update Xerces to Version 2.12.2

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8732/head:pull/8732` \
`$ git checkout pull/8732`

Update a local copy of the PR: \
`$ git checkout pull/8732` \
`$ git pull https://git.openjdk.java.net/jdk pull/8732/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8732`

View PR using the GUI difftool: \
`$ git pr show -t 8732`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8732.diff">https://git.openjdk.java.net/jdk/pull/8732.diff</a>

</details>

### CODE REVIEW

**Review by LanceAndersen — COMMENTED** [review]
Hi Joe,

I think  your changes are ok

Are you going to wait until the JCK is updated to either patch or exclude these tests?

It would be good to get the JCK addressed prior to pushing the changes

**Comment by JoeWang-Java** [comment]
Thanks Lance.

Yes, I've notified the JCK team, will wait till they've resolved the issue (or at least put them in the known list). Meanwhile, just want to get the patch reviewed.

File: test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java [hunk 1]
```diff
@@ -31,21 +31,94 @@
...
- * @bug 8149915 8222991
+ * @bug 8149915 8222991 8282280 8144117
  * @library /javax/xml/jaxp/libs /javax/xml/jaxp/unittest
  * @run testng/othervm validation.SchemaTest
- * @summary Test Schema creation
+ * @summary Tests schemas and Schema creation.
  */
-@Listeners({jaxp.library.FilePolicy.class})
 public class SchemaTest {
+    private static final String XSD_8144117 = "<?xml version='1.0'?>\n" +
```
Comment (naotoj) [inline test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java:48]: If it is not from the upstream Xerces test but our own, can we utilize text blocks here?

File: test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java [hunk 2]
```diff
@@ -31,21 +31,94 @@
...
+     */
+    @DataProvider(name = "xsd")
+    Object[][] getXSDString() {
+        return new Object[][]{
+            {XSD_8144117},
+        };
+    }
+
+    /*
+     * DataProvider: valid xsd
```
Comment (naotoj) [inline test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java:79]: This comment is the same as the above provider. Can we change it to a more meaningful one?

**Review by naotoj — COMMENTED** [review]
Changes look good to me. Some minor nits follow.

File: test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java (hunk 1 above)
Comment (JoeWang-Java) [reply test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java:48]: Our source level is still generally 8. The expectation is also that this patch will be backported.

File: test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java (hunk 2 above)
Comment (JoeWang-Java) [reply test/jaxp/javax/xml/jaxp/unittest/validation/SchemaTest.java]: Updated with a bit more description.

**Review by naotoj — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Comment by jfoster-twilio** [comment]
@JoeWang-Java https://github.com/openjdk/jdk/pull/8732/files#diff-559485fb84e5da2f5c0db2e967e6522895fb398da4d2eca9bd0f571461585b01R3224-R3235 introduces a bug where `reportSchemaError(
                        "cvc-complex-type.4",
                        new Object[] { element.rawname, currDecl.fName });` may be emitted twice.

**Pull request closed without merging**